android-kit-base/src/main/java/com/mparticle/kits/MPSideloadedFilters.kt

```diff
@@ -1,9 +1,297 @@
+            KEY_EVENT_TYPES_FILTER,
+            Pair(
+                HashingUtility.hashFilterTypeCommerceEvent(eventType).toString(),
+                EXCLUDING_FILTER_VALUE
+            )
+        )
+        return this
+    }
+
+    /**
+     * Add event name filter
+     *
+     * @param eventType
+     * @param eventName
+     * @return MPSideloadedFilters instance
+     */
+    // Internally use ec as the key
+    fun addEventNameFilter(eventType: MParticle.EventType, eventName: String): MPSideloadedFilters {
+        applyToFiltersNode(
+            KEY_EVENT_NAMES_FILTER,
+            Pair(
+                HashingUtility.hashEvent(eventType, eventName).toString(),
+                EXCLUDING_FILTER_VALUE
+            )
+        )
+        return this
+    }
+
+    /**
+     * Add screen name filter
+     *
+     * @param screenName
+     * @return MPSideloadedFilters instance
+     */
+    fun addScreenNameFilter(screenName: String): MPSideloadedFilters {
+        applyToFiltersNode(
+            KEY_SCREEN_NAME_FILTER,
+            Pair(
+                HashingUtility.hashFilterScreenName(screenName).toString(),
+                EXCLUDING_FILTER_VALUE
+            )
+        )
+        return this
+    }
+
+    /**
+     * Add event attribute filter
+     *
+     * @param eventType
+     * @param eventName
+     * @param customAttributeKey
+     * @return MPSideloadedFilters instance
+     */
+    // Internally use ea as the key
+    fun addEventAttributeFilter(
+        eventType: MParticle.EventType,
+        eventName: String,
+        customAttributeKey: String
+    ): MPSideloadedFilters {
+        applyToFiltersNode(
+            KEY_EVENT_ATTRIBUTES_FILTER,
+            Pair(
+                HashingUtility.hashFilterEventAttributes(eventType, eventName, customAttributeKey)
+                    .toString(),
+                EXCLUDING_FILTER_VALUE
+            )
+        )
+        return this
+    }
+
+    /**
+     * Add Screen attribute filter
+     *
+     * @param screenName
+     * @param customAttributeKey
+     * @return MPSideloadedFilters instance
+     */
+    // Internally use svea as the key
+    fun addScreenAttributeFilter(
+        screenName: String,
+        customAttributeKey: String
+    ): MPSideloadedFilters {
+        applyToFiltersNode(
+            KEY_SCREEN_ATTRIBUTES_FILTER,
+            Pair(
+                HashingUtility.hashFilterScreenName("$screenName$customAttributeKey").toString(),
+                EXCLUDING_FILTER_VALUE
+            )
+        )
+        return this
+    }
+
+    /**
+     * Add user identity filter
+     *
+     * @param userIdentityType
+     * @return MPSideloadedFilters instance
+     */
+    // Internally use uid as the key
+    fun addUserIdentityFilter(userIdentityType: MParticle.IdentityType): MPSideloadedFilters {
+        applyToFiltersNode(
+            KEY_USER_IDENTITY_FILTER,
+            Pair(
+                userIdentityType.value.toString(),
```

Comment:
So even though user identity type isn't actually hashed, for consistency sake, the plan is to include it in the hashing utility class and just return back the value. See the web implementation here: https://github.com/mParticle/mparticle-web-sdk/blob/ea26d51e8c09129d9550f937192208a5368f75e2/src/kitFilterHelper.ts#L25

The idea is to keep it consistent, allow for potential hashing in the future, and self-document that in this case the "hash" is just the value itself.

Reply:
Creating a hashing fun